src/bluesky/protocols.py

```diff
@@ -352,10 +352,13 @@ def collect_pages(self) -> SyncOrAsyncIterator[PartialEventPage]:
         ...
 
 
+T_CONT = TypeVar("T_CONT", contravariant=True)
```

Comment:
I'm always slightly hazy on when we need to define a TypeVar as contravariant, can you provide a snippet where this fails if it isn't defined as contravariant?

Reply:
Truth be told, I don't really see why this would need to be contravariant explicitly... it just fixed the mypy errors:

```
src/bluesky/protocols.py:358: error: Invariant type variable "T" used in protocol where contravariant one is expected  [misc]
src/bluesky/protocols.py:486: error: Invariant type variable "T" used in protocol where contravariant one is expected  [misc]
```

Reply:
If I had to guess it's because the methods defined on the protocol take `T` so you have to declare that subclasses are allowed.